Explorer/Assets/DCL/InWorldCamera/ScreencaptureCamera/ScreenRecorder.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace DCL.InWorldCamera.ScreencaptureCamera
+{
+    public enum RecordingState
+    {
+        UNKNOWN = 0,
+        IDLE = 1,
+        CAPTURING = 2,
+        SCREENSHOT_READY = 3,
+    }
+
+    public class ScreenRecorder
+    {
+        private const int TARGET_FRAME_WIDTH = 1920;
+        private const int TARGET_FRAME_HEIGHT = 1080;
+        public const float FRAME_SCALE = 0.87f; // Defines the scale of the frame in relation to the screen
+
+        private readonly float targetAspectRatio;
+        private readonly RectTransform canvasRectTransform;
+
+        private readonly Texture2D screenshot = new (TARGET_FRAME_WIDTH, TARGET_FRAME_HEIGHT, TextureFormat.RGB24, false);
+
+        private readonly WaitForEndOfFrame waitForEndOfFrame = new ();
+
+        private RenderTexture originalBaseTargetTexture;
+
+        private ScreenFrameData debugTargetScreenFrame;
+
+        public RecordingState State { get; private set; } = RecordingState.IDLE;
+
+        public ScreenRecorder(RectTransform canvasRectTransform)
+        {
+            targetAspectRatio = (float)TARGET_FRAME_WIDTH / TARGET_FRAME_HEIGHT;
+            Debug.Assert(targetAspectRatio != 0, "Target aspect ratio cannot be null");
+
+            this.canvasRectTransform = canvasRectTransform;
+        }
+
+        public virtual IEnumerator CaptureScreenshot()
+        {
+            State = RecordingState.CAPTURING;
+
+            yield return waitForEndOfFrame; // for UI to appear on screenshot. Converting to UniTask didn't work :(
+
+            ScreenFrameData currentScreenFrame = CalculateCurrentScreenFrame();
+            (_, float targetRescale) = CalculateTargetScreenFrame(currentScreenFrame);
```

Comment:
This is the only call site for CalculateTargetScreenFrame, so you never use the first half of the tuple, so why compute it at all?